compiler/ilgen/IlBuilder.cpp

```diff
@@ -770,6 +770,34 @@ IlBuilder::VectorLoadAt(TR::IlType *dt, TR::IlValue *address)
    return returnValue;
    }
 
+/**
+ * @brief IlBuilder::StructFieldAddress IlGen to get the address of a struct field
+ * @param structName is the name of the struct type the field belongs to
+ * @param fieldName is the name of the field itself
+ * @param obj is an IlValue representing the instance (address) of the struct
+ * @return IlValue representing the address of the field in the struct
+ */
+TR::IlValue *
+IlBuilder::StructFieldAddress(const char* structName, const char* fieldName, TR::IlValue* obj) {
+   auto offset = typeDictionary()->OffsetOf(structName, fieldName);
+   auto ptype = typeDictionary()->PointerTo(typeDictionary()->GetFieldType(structName, fieldName));
+   auto addr = Add(obj, ConstInt64(offset));
+   return ConvertTo(ptype, addr);
+}
+
+/**
+ * @brief IlBuilder::StructFieldAddress IlGen to get the address of a union field
+ * @param unionName is the name of the union type the field belongs to
+ * @param fieldName is the name of the field itself
+ * @param obj is an IlValue representing the instance (address) of the union
+ * @return IlValue representing the address of the field in the union
+ */
+TR::IlValue *
+IlBuilder::UnionFieldAddress(const char* unionName, const char* fieldName, TR::IlValue* obj) {
+   auto ptype = typeDictionary()->PointerTo(typeDictionary()->UnionFieldType(unionName, fieldName));
```

Comment:
In the case of unions, all fields have an offset of 0. So, it's enough to simply cast the pointer to the union to a pointer to the field type; no calculation is require.